### PR TITLE
ci: fix gh actions' warnings

### PR DESCRIPTION
**Pull request by aciba90**
## Proposed Commit Message

```
ci: fix gh actions' warnings

Upgrade to actions/checkout@v3 and migrate to Environment Files instead of `set-output`.
```

## Desired commit type

 - [x] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [ ] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

File: .github/workflows/ci-integration.yaml [hunk 1]
```diff
@@ -84,7 +84,7 @@ jobs:
         if: matrix.platform == 'lxd' || matrix.platform == 'vm'
         run: sudo lxd init --auto
       - name: Git checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Retieve debs
```
Comment (CalvoM) [inline .github/workflows/ci-integration.yaml:88]: I know it is still a WIP, but let me request a tiny change. Please change to `Retrieve`.

Comment (aciba90) [inline .github/workflows/ci-integration.yaml:88]:
```suggestion
      - name: Retrieve debs
```

Comment (aciba90) [reply .github/workflows/ci-integration.yaml]: Done.

File: .github/workflows/check-secrets-available.yaml [hunk 2]
```diff
@@ -22,4 +22,4 @@ jobs:
         env:
           SECRET_TO_CHECK: '${{ secrets.SECRET_TO_CHECK }}'
         if: ${{ env.SECRET_TO_CHECK != '' }}
-        run: echo "::set-output name=has-secrets::true"
+        run: echo "has-secrets=true" >> $GITHUB_OUTPUT
```
Comment (lucasmoura) [inline .github/workflows/check-secrets-available.yaml:25]: @aciba90 do you have a reference for this change ? I don't even a deep knowledge on github actions.
Also, @paride, if you can have a look on it as well

Comment (aciba90) [reply .github/workflows/check-secrets-available.yaml:25]: Sure. One can see these warnings in any executed integration job, for example: https://github.com/canonical/ubuntu-advantage-client/actions/runs/3839211604

This one is:

```
Check secrets / Check secret
The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

and the other one:
```
Packaging (jammy)
Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout@v2
```

**Comment by aciba90** [comment]
Reopen to trigger clean build.

**Comment by aciba90** [comment]
Aside from CI instabilities, this PR is ready to be review.

File: .github/workflows/check-secrets-available.yaml (hunk 2 above)
Comment (paride) [reply .github/workflows/check-secrets-available.yaml:25]: Looks good, thanks!

**Review by paride — APPROVED** [review]

**Pull request merged**